Applications/Spire/Source/Ui/ColorConversion.cpp

```diff
@@ -50,13 +50,46 @@ namespace {
   }
 }
 
+OklabColor::OklabColor()
+  : m_l(-1),
+    m_a(-1),
+    m_b(-1) {}
+
+OklabColor::OklabColor(double l, double a, double b)
+  : m_l(l),
+    m_a(a),
+    m_b(b) {}
+
+bool OklabColor::is_valid() const {
+  if(m_l < 0 || m_l > 1.0 || m_a < -0.4 || m_a > 0.4 ||
+      m_b < -0.4 || m_b > 0.4) {
+    return false;
+  }
+  return true;
+}
+
+OklchColor::OklchColor()
+  : m_l(-1),
+    m_c(-1),
+    m_h(0) {}
+
+OklchColor::OklchColor(double l, double c, double h)
+  : m_l(l),
+    m_c(c),
+    m_h(h) {}
+
+bool OklchColor::is_valid() const {
+  if(m_l < 0 || m_l > 1.0 || m_c < 0 || m_c > 0.4) {
```

Comment:
m_h can be any value?

Reply:
Yes, because m_h represents the hue angle of a color and there is no limit for the angle value.
https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/oklch

I will add some unit tests for it.